src/components/tooltip/Tooltip.tsx

```diff
@@ -217,10 +218,17 @@ class Tooltip extends React.Component<TooltipProps, State> {
             stopBubble,
             text,
             theme,
+            intl,
         } = this.props;
 
-        const tooltipText =
-            text && typeof text === 'object' ? (text as React.ReactElement).props.defaultMessage : String(text);
+        const textElement = text as React.ReactElement;
```

Comment:
ReactElement is not the same as ReactNode, so this cast is masking the real possible types for the `text` prop